packages/graphql-hooks/src/useClientRequest.ts

```diff
@@ -82,18 +100,25 @@ function useClientRequest(query, initialOpts = {}) {
 
   const contextClient = React.useContext(ClientContext)
   const client = initialOpts.client || contextClient
+
+  if (client === null) {
+    throw new Error(
+      'A client must be provided in order to use the useClientRequest hook.'
+    )
+  }
+
   const isMounted = React.useRef(true)
-  const activeCacheKey = React.useRef(null)
+  const activeCacheKey = React.useRef<CacheKeyObject | null>(null)
   const operation = {
     query,
     variables: initialOpts.variables,
     operationName: initialOpts.operationName,
     persisted: initialOpts.persisted
-  }
+  } as any
 
   if (
     initialOpts.persisted ||
-    (client.useGETForQueries && !initialOpts.isMutation)
+    (client?.useGETForQueries && !initialOpts.isMutation)
```

Comment:
if we assert the client is there, why do we need a conditional chaining here? (and below)